### PR TITLE
Clarify docstring of DataSet.export about the name elements

### PR DESCRIPTION
**Pull request by astafan8**

### CODE REVIEW

File: qcodes/dataset/data_set_protocol.py [hunk 1]
```diff
@@ -328,9 +328,13 @@ def export(
         path: str | None = None,
         prefix: str | None = None,
     ) -> None:
-        """Export data to disk with file name {prefix}{run_id}.{ext}.
-        Values for the export type, path and prefix can also be set in the "dataset"
-        section of qcodes config.
+        """Export data to disk with file name {prefix}{name_elements}.{ext}.
```
Comment (jenshnielsen) [inline qcodes/dataset/data_set_protocol.py:331]:
```suggestion
        """Export data to disk with file name `{prefix}{name_elements}.{ext}`.
```

perhaps

File: qcodes/dataset/data_set_protocol.py [hunk 2]
```diff
@@ -328,9 +328,13 @@ def export(
...
+        Name elements are names of dataset object attributes that are taken
+        from the dataset and inserted into the name of the export file, for
+        example if name elements are ``["captured_run_id", "guid"]``, then
+        the file name will be {prefix}{captured_run_id}_{guid}.{ext}.
```
Comment (jenshnielsen) [inline qcodes/dataset/data_set_protocol.py:335]:
```suggestion
        the file name will be `{prefix}{captured_run_id}_{guid}.{ext}`.
```

File: qcodes/dataset/data_set_protocol.py [hunk 3]
```diff
@@ -373,10 +377,13 @@ def _export_data(
         path: str | None = None,
         prefix: str | None = None,
     ) -> str | None:
-        """Export data to disk with file name {prefix}{run_id}.{ext}.
-
-        Values for the export type, path and prefix can also be set in the qcodes
-        "dataset" config.
+        """Export data to disk with file name {prefix}{name_elements}.{ext}.
```
Comment (jenshnielsen) [inline qcodes/dataset/data_set_protocol.py:380]:
```suggestion
        """Export data to disk with file name `{prefix}{name_elements}.{ext}`.
```

File: qcodes/dataset/data_set_protocol.py [hunk 4]
```diff
@@ -373,10 +377,13 @@ def _export_data(
...
+        Name elements are names of dataset object attributes that are taken
+        from the dataset and inserted into the name of the export file, for
+        example if name elements are ``["captured_run_id", "guid"]``, then
+        the file name will be {prefix}{captured_run_id}_{guid}.{ext}.
```
Comment (jenshnielsen) [inline qcodes/dataset/data_set_protocol.py:384]:
```suggestion
        the file name will be `{prefix}{captured_run_id}_{guid}.{ext}`.
```

**Review by jenshnielsen — APPROVED** [review]

**Comment by astafan8** [comment]
bors merge

**Comment by jenshnielsen** [comment]
Bors merge

**Comment by jenshnielsen** [comment]
bors merge

**Pull request merged**